### DISCUSSION TITLE
Enrich `good first issue`

### DISCUSSION THREAD

**Discussion by panno8M** [godot-nim/.github#12] [General]
@xix-xeaon @insomniacUNDERSCORElemon 
In [the PR](https://github.com/godot-nim/gdext-nim/pull/118#issuecomment-2440195763) I mentioned earlier, I noticed that this project has an unusually high hurdle for new contributors to get involved. I think we have no issues lined up now for newcomers.
I am looking for tasks that I could mark as `good first issue`, but in no way I guess I am used to the status quo. From now on, I will actively try to clarify these minor modifications as an issue.

In the meantime, is there any  `good first issue`?

**Comment by insomniacUNDERSCORElemon** [comment]
I imagine this is just a mix of early adoption, Nim being somewhat niche, plus everyone having their own skillset and interests. Deep understanding of Godot, Nim, and git is a tall order. 

With the stars up to 45 now and getting more potential contributors at all, I'd say that it means the project is on the right track. Just hopefully you get some star players before a big influx of interest.

Wish I could be more helpful, but I'm a bit aimless right now. Gamedev is even less my thing, difficult thinking of viable starter project ideas for even a tiny amount of motivation... even the smallest amount of friction grinds that to a halt. Which is admittedly my problem(s).

**Comment by xix-xeaon** [comment]
I've been going through the project quite a lot, mostly just to figure out how to use it, and update the wiki on that. I was able to contribute a small piece of code, but for anything more significant I still don't understand how it works well enough. For this reason I think documentation is the most important area to improve in order to help people contribute to the project.

First of all, an explanation of the **project architecture** is needed (in a wikipage probably). What are the major parts? How do they work and relate to each other? What are the design decisions and considerations.

Imagine that you have a _student_ who's reasonably proficient in nim but they've **never** created something like a gdextension language binding. What would you need to explain about how you've done it in order for them to be able to _create the same thing themselves_?

Secondly, **document important details** in the source code. Focus on **why** the code is that way, rather than what it does. Again, imagine the student. For most of the code they'll understand **what** it does (just by reading it) but they won't know **why** it was made that way. Some code might also require explanations on what it does if it uses advanced/obscure techniques or works with _"magic numbers"_.

I realize there's a language barrier but from what I've seen I think you're doing fine with the translations =) Any explanations you can add are better than nothing, and you can simply add more, a bit at a time, as I've done with the quick guide.

**Reply by panno8M** [reply]
Thanks for your clear explanation. So you says, It is more meaningful to improve developer documentation and make it easier to develop core functionality than to identify easy issues, isn't it.

I certainly think this is important. To be honest it is not a very interesting task, but I will work on it little by little.

Btw, is it easy to understand that I am using a translator?

**Reply by xix-xeaon** [reply]
It's not apparent that you're using a translator, but I saw somewhere else, you wrote that you do. I had previously noted that you're from Japan (which I find to be a most interesting country), where the statistics note a low level of skill in English.

I'm a teacher (of programming) and this summer during my month long visit to Japan I also visited a few schools. I had a translator from the city's tourism bureau for that, but even her English left a lot to be desired. Her senpai had to use a translator app when she wasn't there. I also went to several meetups and most Japanese natives had to use translator apps for most things.

I studied a bit of Japanese before, just the very very basics (but it was certainly helpful), and it's interesting to note certain patterns in your writing that would never come up in native English but do make perfect sense for Japanese. So it's noticeable that you don't speak native level English, but if you're using a translator or not, I couldn't tell from the text itself. In either case, it's perfectly understandable. You have nothing to worry about.

**Reply by panno8M** [reply]
yeah I'm Japanese; I also like Sweden :)

> I'm a teacher (of programming) and this summer during my month long visit to Japan I also visited a few schools. I had a translator from the city's tourism bureau for that, but even her English left a lot to be desired. Her senpai had to use a translator app when she wasn't there. I also went to several meetups and most Japanese natives had to use translator apps for most things.

Oh you had came to Japan? And a teacher? I'm surprised; I'm a computer-related undergraduate.
Indeed, Japanese people can hardly speak English. Japanese who can speak English literally look radiant in my sight.

> and it's interesting to note certain patterns in your writing that would never come up in native English but do make perfect sense for Japanese. 

That's certainly interesting. More precisely, I use the translation as a *draft*, which may have some influence.

Well, now I lose one excuse for not writing documentation. I'll do my best.
